### PR TITLE
Implement passwordRecoveryReminder modal

### PR DESCRIPTION
**Pull request by kylanhurt**
The purpose of this task is to implement a reminder modal upon new transactions received that urges the user to set up password recovery. The fiat equivalent balance levels for prompting are $20, $200, $2000, $20000, and $200000.

The modal should only prompt once after each one of these levels is passed (assuming password recovery has not yet been set up).

Asana Task: https://app.asana.com/0/361770107085503/498144750192816/f

### CODE REVIEW

File: src/modules/Core/Account/callbacks.js [hunk 1]
```diff
@@ -59,6 +61,10 @@ const makeAccountCallbacks = (dispatch: Dispatch): EdgeAccountCallbacks => {
       dispatch(refreshReceiveAddressRequest(walletId))
       dispatch(newTransactionsRequest(walletId, transactions))
       dispatch(refreshWallet(walletId))
+      // now check if password recovery is set up
```
Comment (wolverineks) [inline src/modules/Core/Account/callbacks.js]: what is this doing?

File: src/modules/Main.ui.js [hunk 2]
```diff
@@ -633,6 +634,7 @@ export default class Main extends Component<Props, State> {
         <AutoLogout />
         <ContactsLoader />
         <PasswordReminderModal />
+        <PasswordRecoveryModalConnector />
```
Comment (wolverineks) [inline src/modules/Main.ui.js]: Rename to `<PasswordRecoveryRemonderModal />`

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import { updateSettings as updateReduxSettings } from '../../Settings/action.js'
+
+export const SHOW_PASSWORD_RECOVERY_MODAL = 'SHOW_PASSWORD_RECOVERY_MODAL'
+export const HIDE_PASSWORD_RECOVERY_MODAL = 'HIDE_PASSWORD_RECOVERY_MODAL'
+
+export const checkPasswordRecovery = () => async (dispatch: Dispatch, getState: GetState) => {
+  const state = getState()
+  const settings = state.ui.settings
+  const account = state.core.account
+  const passwordRecoveryRemindersShown = { ...settings.passwordRecoveryRemindersShown }
```
Comment (wolverineks) [inline src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: why are you creating a new object here?

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  const totalDollars = tallyUpTotalCrypto(state)
```
Comment (wolverineks) [inline src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: how expensive is this function?

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  const isPasswordRecoverySetup = !!account.recoveryKey
+  for (const level in passwordRecoveryRemindersShown) {
+    if (isPasswordRecoverySetup) return
```
Comment (wolverineks) [inline src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: could this be done before entering into the loop?

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+export const HIDE_PASSWORD_RECOVERY_MODAL = 'HIDE_PASSWORD_RECOVERY_MODAL'
+
+export const checkPasswordRecovery = () => async (dispatch: Dispatch, getState: GetState) => {
+  const state = getState()
+  const settings = state.ui.settings
+  const account = state.core.account
+  const passwordRecoveryRemindersShown = { ...settings.passwordRecoveryRemindersShown }
+  const totalDollars = tallyUpTotalCrypto(state)
+  const isPasswordRecoverySetup = !!account.recoveryKey
+  for (const level in passwordRecoveryRemindersShown) {
```
Comment (wolverineks) [inline src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: have you already attempted to use `Array#forEach` for this section?

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    if (bns.lt(totalDollars, level)) return // if balance is not big enough to trigger then exit routine
+    if (passwordRecoveryRemindersShown[level] === true) continue // if it's already been shown then go to higher level
+    // now show the modal
+    dispatch(showPasswordRecoveryModal())
+    passwordRecoveryRemindersShown[level] = true
+    const syncedSettings = await getSyncedSettingsAsync(account)
+    const newSyncedSettings = { ...syncedSettings, passwordRecoveryRemindersShown }
+    await setSyncedSettingsAsync(account, newSyncedSettings)
+    dispatch(
+      updateReduxSettings({
```
Comment (wolverineks) [inline src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: what is this?

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  const totalDollars = tallyUpTotalCrypto(state)
+  const isPasswordRecoverySetup = !!account.recoveryKey
+  for (const level in passwordRecoveryRemindersShown) {
+    if (isPasswordRecoverySetup) return
+    if (bns.lt(totalDollars, level)) return // if balance is not big enough to trigger then exit routine
+    if (passwordRecoveryRemindersShown[level] === true) continue // if it's already been shown then go to higher level
+    // now show the modal
+    dispatch(showPasswordRecoveryModal())
+    passwordRecoveryRemindersShown[level] = true
+    const syncedSettings = await getSyncedSettingsAsync(account)
```
Comment (wolverineks) [inline src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: all of this is in `ACCOUNT_API` already
have you already attempted to use it?

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+export const hidePasswordRecoveryModal = () => {
+  return {
+    type: HIDE_PASSWORD_RECOVERY_MODAL
+  }
+}
+
+export const onGoToPasswordRecoveryScene = () => (dispatch: Dispatch, getState: GetState) => {
+  dispatch(hidePasswordRecoveryModal())
+  Actions[RECOVER_PASSWORD]()
```
Comment (wolverineks) [inline src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: please rename this constant to match the constant naming convention

File: src/modules/utils.js [hunk 10]
```diff
@@ -430,7 +430,7 @@ export const getTimeInMinutes = (params: { measurement: string, value: number })
   const { measurement, value } = params
   const measurementStrategies = {
     seconds (v) {
-      const val = Math.round(v / 60 * 100) / 100
+      const val = Math.round((v / 60) * 100) / 100
```
Comment (wolverineks) [inline src/modules/utils.js]: what is this?

File: src/modules/utils.js [hunk 11]
```diff
@@ -567,3 +567,56 @@ export const isEdgeLogin = (data: string) => {
 
   return EDGE_LOGIN_REG_EXP.test(data)
 }
+
+export const tallyUpTotalCrypto = (state: State) => {
```
Comment (wolverineks) [inline src/modules/utils.js]: how expensive is this function?

File: src/modules/Core/Account/callbacks.js [hunk 12]
```diff
@@ -59,6 +61,10 @@ const makeAccountCallbacks = (dispatch: Dispatch): EdgeAccountCallbacks => {
...
+      const finalTxIndex = transactions.length - 1
+      if (bns.gt(transactions[finalTxIndex].nativeAmount, '0')) dispatch(checkPasswordRecovery())
```
Comment (wolverineks) [inline src/modules/Core/Account/callbacks.js]: `utils.isIncomingTransaction`

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js (hunk 3 above)
Comment (wolverineks) [inline src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: check if password recovery is setup,
if it is already setup, stop working as soon as possible

all code after that point should assume that password recovery is not setup

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js (hunk 8 above)
Comment (wolverineks) [reply src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: make a SETTINGS_API function that does this

File: src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js (hunk 6 above)
Comment (wolverineks) [reply src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js]: `Object.keys(settings).forEach(key => settings.key)`

File: src/modules/utils.js (hunk 11 above)
Comment (kylanhurt) [reply src/modules/utils.js]: Not very, in the low 10's of milliseconds.

File: src/modules/utils.js (hunk 10 above)
Comment (kylanhurt) [reply src/modules/utils.js]: Automatic formatting by `yarn format`

File: src/modules/Core/Account/settings.js [hunk 13]
```diff
@@ -11,6 +11,14 @@ export const CORE_DEFAULTS = {
   pinMode: false
 }
 
+export const PASSWORD_RECOVER_REMINDERS_SHOWN = {
```
Comment (wolverineks) [inline src/modules/Core/Account/settings.js]: typo?

File: src/modules/UI/Settings/action.js [hunk 14]
```diff
@@ -48,15 +48,21 @@ export const ADD_CURRENCY_PLUGIN = PREFIX + 'ADD_CURRENCY_PLUGIN'
 // tokens
 export const SET_CUSTOM_TOKENS = PREFIX + 'SET_CUSTOM_TOKENS'
 
-export const updateSettings = (settings: Object) => ({
-  type: UPDATE_SETTINGS,
-  data: { settings }
-})
+export const updateSettings = (settings: Object) => {
+  console.log('updateSettings: ', settings)
```
Comment (wolverineks) [inline src/modules/UI/Settings/action.js]: remove console.log

File: src/modules/UI/Settings/action.js [hunk 15]
```diff
@@ -48,15 +48,21 @@ export const ADD_CURRENCY_PLUGIN = PREFIX + 'ADD_CURRENCY_PLUGIN'
...
+    data: { settings }
+  }
+}
 
-export const loadSettings = (settings: Object) => ({
-  type: LOAD_SETTINGS,
-  data: { settings }
-})
+export const loadSettings = (settings: Object) => {
+  console.log('loadSettings: ', settings)
```
Comment (wolverineks) [inline src/modules/UI/Settings/action.js]: remove console.log

File: src/modules/UI/Settings/reducer.js [hunk 16]
```diff
@@ -8,6 +8,7 @@ import type { CustomTokenInfo } from '../../../types'
 import { CORE_DEFAULTS, LOCAL_ACCOUNT_DEFAULTS, SYNCED_ACCOUNT_DEFAULTS } from '../../Core/Account/settings.js'
 import { SEND_LOGS_FAILURE, SEND_LOGS_PENDING, SEND_LOGS_REQUEST, SEND_LOGS_SUCCESS } from '../../Logs/action'
 import type { Action } from '../../ReduxTypes'
+import { UPDATE_SHOW_PASSWORD_RECOVER_REMINDER_MODAL } from '../components/PasswordRecoveryReminderModal/PasswordRecoveryReminderModalActions.js'
```
Comment (wolverineks) [inline src/modules/UI/Settings/reducer.js]: typo?

File: src/modules/UI/Settings/reducer.js [hunk 17]
```diff
@@ -530,6 +540,16 @@ export const settingsLegacy = (state: SettingsState = initialState, action: Acti
       }
     }
 
+    case UPDATE_SHOW_PASSWORD_RECOVER_REMINDER_MODAL: {
```
Comment (wolverineks) [inline src/modules/UI/Settings/reducer.js]: typo?

**Pull request merged**